tests/integration/_experimental/object_detection/test_evaluator_single_class_test_case_metrics.py

```diff
@@ -0,0 +1,904 @@
+        "iou=0.33 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "iou=0.33 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "iou=0.33 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.0",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.0",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "iou=0.5 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(
+        Objects=1,
+        Inferences=1,
+        TP=1,
+        FN=0,
+        FP=0,
+        Precision=1.0,
+        Recall=1.0,
+        F1=1.0,
+        AP=0.07494252873563219,
```

Comment:
Is `AP` correct there?

Reply:
It's weird to me that `AP` changes as the threshold changes (see the next 3 tests)

Reply:
Agreed, originally the cache of matchings by test case name would follow through in an evaluator, but it makes more sense for each test to be completely independent.